src/util/source_location.h

```diff
@@ -151,6 +151,19 @@ class source_locationt:public irept
     return set(ID_basic_block_covered_lines, covered_lines);
   }
 
+  /// Add comment to source location as originating from removed virtual call
+  void set_java_removed_virtual_call()
+  {
+    set(ID_C_java_removed_virtual_call, true);
+  }
+
+  /// Get information about whether call originated from a removed virtual
+  /// function call
+  bool get_java_removed_virtual_call() const
+  {
+    return get_bool(ID_C_java_removed_virtual_call);
+  }
+
```

Comment:
I'm failing to parse the commit message for this one, and I'm not too happy to have Java-specific APIs enter something as general as a `source_locationt`? You might play the same game as is done with various other irept-children and introduce a `java_source_locationt` that has those extra functions?

Reply:
@tautschnig I don't mind, but please note that `source_locationt` already has `get_java_bytecode_index`

Reply:
Maybe you can fix that all in one go?

Reply:
As this is a fairly large refactoring I've raised an issue (#1824) to tackle this separately.